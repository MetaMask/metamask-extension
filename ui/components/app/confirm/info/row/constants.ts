export const TEST_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';

export enum RowAlertKey {
  EstimatedFee = 'estimatedFee',
  SigningInWith = 'signingInWith',
  RequestFrom = 'requestFrom',
  Resimulation = 'resimulation',
  Speed = 'speed',
}

export enum AlertActionKey {
  Buy = 'buy',
  ShowAdvancedGasFeeModal = 'showAdvancedGasModal',
  ShowGasFeeModal = 'showGasFeeModal',
}
