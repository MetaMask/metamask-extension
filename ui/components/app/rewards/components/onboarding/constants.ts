export const REWARDS_ONBOARD_OPTIN_LEGAL_LEARN_MORE_URL =
  'https://support.metamask.io/manage-crypto/metamask-rewards';

export const REWARDS_ONBOARD_TERMS_URL = 'https://go.metamask.io/rewards-terms';
