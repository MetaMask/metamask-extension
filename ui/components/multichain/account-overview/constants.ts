export const FUND_SLIDE = {
  id: 'fund',
  title: 'slideFundWalletTitle',
  description: 'slideFundWalletDescription',
  image: './images/slide-fund-icon.svg',
  href: 'https://portfolio.metamask.io/buy/build-quote',
};

///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
export const BRIDGE_SLIDE = {
  id: 'bridge',
  title: 'slideBridgeTitle',
  description: 'slideBridgeDescription',
  image: './images/slide-bridge-icon.svg',
};
///: END:ONLY_INCLUDE_IF

export const CARD_SLIDE = {
  id: 'card',
  title: 'slideDebitCardTitle',
  description: 'slideDebitCardDescription',
  image: './images/slide-card-icon.svg',
  href: 'https://portfolio.metamask.io/card',
};

export const CASH_SLIDE = {
  id: 'cash',
  title: 'slideCashOutTitle',
  description: 'slideCashOutDescription',
  image: './images/slide-sell-icon.svg',
  href: 'https://portfolio.metamask.io/sell',
};

export const ZERO_BALANCE = '0x00';
