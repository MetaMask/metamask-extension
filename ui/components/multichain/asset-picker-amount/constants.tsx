export const LARGE_SYMBOL_LENGTH = 7;
