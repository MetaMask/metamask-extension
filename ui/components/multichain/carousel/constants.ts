export const MARGIN_VALUES = {
  SLIDE_BOTTOM: '40px',
  CONTAINER_SIDE: '16px',
  SLIDE_GAP: '2%',
  ZERO: '0',
};

export const WIDTH_VALUES = {
  FULL_WIDTH: '100%',
  STANDARD_SLIDE: '98%',
  FIRST_SLIDE: '94%',
};

export const BANNER_STYLES = {
  HEIGHT: '59px',
};

export const MAX_SLIDES = 5;
