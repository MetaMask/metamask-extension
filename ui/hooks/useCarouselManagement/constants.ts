export const REMOTE_MODE_SLIDE = {
  id: 'remoteMode',
  title: 'slideRemoteModeTitle',
  description: 'slideRemoteModeDescription',
  image: './images/slide-cold-storage.svg',
  href: '/home.html#remote',
};

///: BEGIN:ONLY_INCLUDE_IF(solana)
export const SOLANA_SLIDE = {
  id: 'solana',
  title: 'slideSolanaTitle',
  description: 'slideSolanaDescription',
  image: './images/slide-solana-icon.svg',
};
///: END:ONLY_INCLUDE_IF

export const SMART_ACCOUNT_UPGRADE_SLIDE = {
  id: 'smartAccountUpgrade',
  title: 'slideSmartAccountUpgradeTitle',
  description: 'slideSmartAccountUpgradeDescription',
  image: './images/smart-account-upgrade.png',
};

export const SWEEPSTAKES_SLIDE = {
  id: 'sweepStake',
  title: 'slideSweepStakeTitle',
  description: 'slideSweepStakeDescription',
  image: './images/sweepstakes.png',
  href: 'https://portfolio.metamask.io/explore/nfts?event=012e19e2',
};

export const FUND_SLIDE = {
  id: 'fund',
  title: 'slideFundWalletTitle',
  description: 'slideFundWalletDescription',
  image: './images/slide-fund-icon.svg',
  href: 'https://portfolio.metamask.io/buy/build-quote',
};

export const BRIDGE_SLIDE = {
  id: 'bridge',
  title: 'slideBridgeTitle',
  description: 'slideBridgeDescription',
  image: './images/slide-bridge-icon.svg',
};

export const CARD_SLIDE = {
  id: 'card',
  title: 'slideDebitCardTitle',
  description: 'slideDebitCardDescription',
  image: './images/slide-card-icon.svg',
  href: 'https://portfolio.metamask.io/card',
};

export const CASH_SLIDE = {
  id: 'cash',
  title: 'slideCashOutTitle',
  description: 'slideCashOutDescription',
  image: './images/slide-sell-icon.svg',
  href: 'https://portfolio.metamask.io/sell',
};

export const MULTI_SRP_SLIDE = {
  id: 'multiSrp',
  title: 'slideMultiSrpTitle',
  description: 'slideMultiSrpDescription',
  image: './images/slide-multi-srp-icon.svg',
  href: '/home.html#import-srp',
};

export const BACKUPANDSYNC_SLIDE = {
  id: 'backupAndSync',
  title: 'backupAndSyncSlideTitle',
  description: 'backupAndSyncSlideDescription',
  image: './images/slide-backup-and-sync-icon.png',
};

export const ZERO_BALANCE = '0x0';

export const SWEEPSTAKES_START = new Date('2025-04-09T00:00:00Z');
export const SWEEPSTAKES_END = new Date('2025-04-28T23:59:59Z');
