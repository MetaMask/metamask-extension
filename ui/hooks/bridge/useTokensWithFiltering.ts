import { useCallback } from 'react';
import { useSelector } from 'react-redux';
import { isEqual } from 'lodash';
import { ChainId } from '@metamask/controller-utils';
import { Hex } from '@metamask/utils';
import { useParams } from 'react-router-dom';
import { zeroAddress } from 'ethereumjs-util';
import {
  getAllDetectedTokensForSelectedAddress,
  getCurrentCurrency,
  getSelectedInternalAccountWithBalance,
  getTokenExchangeRates,
} from '../../selectors';
import { getConversionRate } from '../../ducks/metamask/metamask';
import { SwapsTokenObject } from '../../../shared/constants/swaps';
import {
  AssetWithDisplayData,
  ERC20Asset,
  NativeAsset,
} from '../../components/multichain/asset-picker-amount/asset-picker-modal/types';
import { AssetType } from '../../../shared/constants/transaction';
import { isNativeAddress } from '../../pages/bridge/utils/quote';
import { CHAIN_ID_TOKEN_IMAGE_MAP } from '../../../shared/constants/network';
import { getCurrentChainId } from '../../../shared/modules/selectors/networks';
import { Token } from '../../components/app/assets/token-list/token-list';
import { useMultichainBalances } from '../useMultichainBalances';

type FilterPredicate = (
  symbol: string,
  address?: string,
  tokenChainId?: string,
) => boolean;

/**
 * Returns a token list generator that filters and sorts tokens in this order
 * - matches URL token parameter
 * - matches search query
 * - highest balance in selected currency
 * - detected tokens (with balance)
 * - popularity
 * - all other tokens
 *
 * @param tokenList - a mapping of token addresses in the selected chainId to token metadata from the bridge-api
 * @param topTokens - a list of top tokens from the swap-api
 * @param tokenAddressAllowlistByChainId - a mapping of all supported chainIds to a Set of allowed token addresses
 * @param chainId - the selected src/dest chainId
 */
export const useTokensWithFiltering = (
  tokenList: Record<string, SwapsTokenObject>,
  topTokens: { address: string }[],
  tokenAddressAllowlistByChainId: Record<string, Set<string>>,
  chainId?: ChainId | Hex,
) => {
  const { token: tokenAddressFromUrl } = useParams();
  const allDetectedTokens: Record<string, Token[]> = useSelector(
    getAllDetectedTokensForSelectedAddress,
  );

  const { balance } = useSelector(getSelectedInternalAccountWithBalance);

  const tokenConversionRates = useSelector(getTokenExchangeRates, isEqual);
  const conversionRate = useSelector(getConversionRate);
  const currentCurrency = useSelector(getCurrentCurrency);
  const currentChainId = useSelector(getCurrentChainId);

  const { assetsWithBalance: multichainTokensWithBalance } =
    useMultichainBalances();

  // This transforms the token object from the bridge-api into the format expected by the AssetPicker
  const buildTokenData = (
    token?: SwapsTokenObject,
  ): AssetWithDisplayData<NativeAsset | ERC20Asset> | undefined => {
    if (!chainId || !token) {
      return undefined;
    }
    // Only tokens on the active chain are processed here here
    const sharedFields = { ...token, chainId };

    if (isNativeAddress(token.address)) {
      return {
        ...sharedFields,
        type: AssetType.native,
        address: zeroAddress(),
        image:
          CHAIN_ID_TOKEN_IMAGE_MAP[
            chainId as keyof typeof CHAIN_ID_TOKEN_IMAGE_MAP
          ],
        balance: currentChainId === chainId ? balance : '',
        string: currentChainId === chainId ? balance : '',
      };
    }

    return {
      ...sharedFields,
      type: AssetType.token,
      image: token.iconUrl,
      // Only tokens with 0 balance are processed here so hardcode empty string
      balance: '',
      string: undefined,
      address: token.address || zeroAddress(),
    };
  };

  // This returns whether the token is blocked by any of the supported chainIds
  const isTokenBlocked = (tokenAddress: string, tokenChainId: string) =>
    !tokenAddressAllowlistByChainId[tokenChainId]?.has(
      tokenAddress.toLowerCase(),
    );

  // shouldAddToken is a filter condition passed in from the AssetPicker that determines whether a token should be included
  const filteredTokenListGenerator = useCallback(
    (shouldAddToken: FilterPredicate) =>
      (function* (): Generator<
        AssetWithDisplayData<NativeAsset> | AssetWithDisplayData<ERC20Asset>
      > {
        // If a token address is in the URL (e.g. from a deep link), yield that token first
        if (tokenAddressFromUrl) {
          const token =
            tokenList?.[tokenAddressFromUrl] ??
            tokenList?.[tokenAddressFromUrl.toLowerCase()];
          if (
            shouldAddToken(token.symbol, token.address ?? undefined, chainId)
          ) {
            const tokenWithData = buildTokenData(token);
            if (tokenWithData) {
              yield tokenWithData;
            }
          }
        }

        // Yield multichain tokens with balances and are not blocked
        for (const token of multichainTokensWithBalance) {
          if (
            shouldAddToken(
              token.symbol,
              token.address ?? undefined,
              token.chainId,
            ) &&
            (token.address
              ? !isTokenBlocked(token.address, token.chainId)
              : true)
          ) {
            // If there's no address, set it to the native address in swaps/bridge
            yield { ...token, address: token.address || zeroAddress() };
          }
        }

        // Yield all detected tokens for all supported chains
        for (const token of Object.values(allDetectedTokens).flat()) {
          if (
            shouldAddToken(
              token.symbol,
              token.address ?? undefined,
              token.chainId,
            ) &&
            (token.address
              ? !isTokenBlocked(token.address, token.chainId)
              : true)
          ) {
            yield {
              ...token,
              type: AssetType.token,
              // Balance is not 0 but is not in the data so hardcode 0
              // If a detected token is selected useLatestBalance grabs the on-chain balance
              balance: '',
              string: undefined,
            };
          }
        }

        // Yield topTokens from selected chain
        for (const token_ of topTokens) {
          const matchedToken =
            tokenList?.[token_.address] ??
            tokenList?.[token_.address.toLowerCase()];
          if (
            matchedToken &&
            shouldAddToken(
              matchedToken.symbol,
              matchedToken.address ?? undefined,
              chainId,
            )
          ) {
            const token = buildTokenData(matchedToken);
            if (token) {
              yield token;
            }
          }
        }

        // Yield other tokens from selected chain
        for (const token_ of Object.values(tokenList)) {
          if (
            token_ &&
            shouldAddToken(token_.symbol, token_.address ?? undefined, chainId)
          ) {
            const token = buildTokenData(token_);
            if (token) {
              yield token;
            }
          }
        }
      })(),
    [
      multichainTokensWithBalance,
      topTokens,
      tokenConversionRates,
      conversionRate,
      currentCurrency,
      chainId,
      tokenList,
      tokenAddressFromUrl,
      allDetectedTokens,
    ],
  );

  return filteredTokenListGenerator;
};
