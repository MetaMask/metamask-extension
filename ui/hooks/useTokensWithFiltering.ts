import { useCallback, useMemo } from 'react';
import { useSelector } from 'react-redux';
import { isEqual } from 'lodash';
import { ChainId, hexToBN } from '@metamask/controller-utils';
import { Hex } from '@metamask/utils';
import {
  getAllTokens,
  getCurrentCurrency,
  getSelectedInternalAccountWithBalance,
  getShouldHideZeroBalanceTokens,
  getTokenExchangeRates,
} from '../selectors';
import { getConversionRate } from '../ducks/metamask/metamask';
import {
  SWAPS_CHAINID_DEFAULT_TOKEN_MAP,
  SwapsTokenObject,
  TokenBucketPriority,
} from '../../shared/constants/swaps';
import { getValueFromWeiHex } from '../../shared/modules/conversion.utils';
import { EtherDenomination } from '../../shared/constants/common';
import {
  AssetWithDisplayData,
  ERC20Asset,
  NativeAsset,
  TokenWithBalance,
} from '../components/multichain/asset-picker-amount/asset-picker-modal/types';
import { AssetType } from '../../shared/constants/transaction';
import { isSwapsDefaultTokenSymbol } from '../../shared/modules/swaps.utils';
import { useTokenTracker } from './useTokenTracker';
import { getRenderableTokenData } from './useTokensToSearch';

/*
 * Returns a token list generator that filters and sorts tokens based on
 * query match, balance/popularity, all other tokens
 */
export const useTokensWithFiltering = (
  tokenList: Record<string, SwapsTokenObject>,
  topTokens: { address: string }[],
  sortOrder: TokenBucketPriority = TokenBucketPriority.owned,
  chainId?: ChainId | Hex,
) => {
  // Only includes non-native tokens
  const allDetectedTokens = useSelector(getAllTokens);
  const { address: selectedAddress, balance: balanceOnActiveChain } =
    useSelector(getSelectedInternalAccountWithBalance);

  const allDetectedTokensForChainAndAddress = chainId
    ? allDetectedTokens?.[chainId]?.[selectedAddress] ?? []
    : [];

  const shouldHideZeroBalanceTokens = useSelector(
    getShouldHideZeroBalanceTokens,
  );
  const {
    tokensWithBalances: erc20TokensWithBalances,
  }: { tokensWithBalances: TokenWithBalance[] } = useTokenTracker({
    tokens: allDetectedTokensForChainAndAddress,
    address: selectedAddress,
    hideZeroBalanceTokens: Boolean(shouldHideZeroBalanceTokens),
  });

  const tokenConversionRates = useSelector(getTokenExchangeRates, isEqual);
  const conversionRate = useSelector(getConversionRate);
  const currentCurrency = useSelector(getCurrentCurrency);

  const sortedErc20TokensWithBalances = useMemo(
    () =>
      erc20TokensWithBalances.toSorted(
        (a, b) => Number(b.string) - Number(a.string),
      ),
    [erc20TokensWithBalances],
  );

  const filteredTokenListGenerator = useCallback(
    (shouldAddToken: (symbol: string, address?: string) => boolean) => {
      const buildTokenData = (
        token: SwapsTokenObject,
      ):
        | AssetWithDisplayData<NativeAsset>
        | AssetWithDisplayData<ERC20Asset>
        | undefined => {
        if (chainId && shouldAddToken(token.symbol, token.address)) {
          return getRenderableTokenData(
            {
              ...token,
              type: isSwapsDefaultTokenSymbol(token.symbol, chainId)
                ? AssetType.native
                : AssetType.token,
              image: token.iconUrl,
            },
            tokenConversionRates,
            conversionRate,
            currentCurrency,
            chainId,
            tokenList,
          );
        }
        return undefined;
      };

      return (function* (): Generator<
        AssetWithDisplayData<NativeAsset> | AssetWithDisplayData<ERC20Asset>
      > {
        const balance = hexToBN(balanceOnActiveChain);
        const srcBalanceFields =
          sortOrder === TokenBucketPriority.owned
            ? {
                balance: balanceOnActiveChain,
                string: getValueFromWeiHex({
                  value: balance,
                  numberOfDecimals: 4,
                  toDenomination: EtherDenomination.ETH,
                }),
              }
            : {};
        const nativeToken = buildTokenData({
          ...SWAPS_CHAINID_DEFAULT_TOKEN_MAP[
            chainId as keyof typeof SWAPS_CHAINID_DEFAULT_TOKEN_MAP
          ],
          ...srcBalanceFields,
        });
        if (nativeToken) {
          yield nativeToken;
        }

        if (sortOrder === TokenBucketPriority.owned) {
          for (const tokenWithBalance of sortedErc20TokensWithBalances) {
            const cachedTokenData =
              tokenWithBalance.address &&
              tokenList &&
              (tokenList[tokenWithBalance.address] ??
                tokenList[tokenWithBalance.address.toLowerCase()]);
            if (cachedTokenData) {
              const combinedTokenData = buildTokenData({
                ...tokenWithBalance,
                ...(cachedTokenData ?? {}),
              });
              if (combinedTokenData) {
                yield combinedTokenData;
              }
            }
          }
        }

        for (const topToken of topTokens) {
          const tokenListItem =
            tokenList?.[topToken.address] ??
            tokenList?.[topToken.address.toLowerCase()];
          if (tokenListItem) {
            const tokenWithTokenListData = buildTokenData(tokenListItem);
            if (tokenWithTokenListData) {
              yield tokenWithTokenListData;
            }
          }
        }

        for (const token of Object.values(tokenList)) {
          const tokenWithTokenListData = buildTokenData(token);
          if (tokenWithTokenListData) {
            yield tokenWithTokenListData;
          }
        }
      })();
    },
    [
      balanceOnActiveChain,
      sortedErc20TokensWithBalances,
      topTokens,
      tokenConversionRates,
      conversionRate,
      currentCurrency,
      chainId,
      tokenList,
    ],
  );

  return filteredTokenListGenerator;
};
