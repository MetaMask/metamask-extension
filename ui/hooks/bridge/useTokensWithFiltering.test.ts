import { renderHookWithProvider } from '../../../test/lib/render-helpers';
import { createBridgeMockStore } from '../../../test/jest/mock-store';
import { STATIC_MAINNET_TOKEN_LIST } from '../../../shared/constants/tokens';
import { SWAPS_CHAINID_DEFAULT_TOKEN_MAP } from '../../../shared/constants/swaps';
import { CHAIN_IDS } from '../../../shared/constants/network';
import { useTokensWithFiltering } from './useTokensWithFiltering';

const mockUseTokenTracker = jest
  .fn()
  .mockReturnValue({ tokensWithBalances: [] });
jest.mock('../useTokenTracker', () => ({
  useTokenTracker: () => mockUseTokenTracker(),
}));

const NATIVE_TOKEN = SWAPS_CHAINID_DEFAULT_TOKEN_MAP[CHAIN_IDS.MAINNET];

describe('useTokensWithFiltering', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return all tokens when chainId === activeChainId, sorted by balance', () => {
    const mockStore = createBridgeMockStore({
      metamaskStateOverrides: {
        completedOnboarding: true,
        allDetectedTokens: {
          '0x1': {
            '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': [
              {
                address: '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984',
                decimals: 6,
              }, // USDC
            ],
          },
        },
      },
    });
    const { result } = renderHookWithProvider(
      () =>
        useTokensWithFiltering(
          {
            [NATIVE_TOKEN.address]: NATIVE_TOKEN,
            ...STATIC_MAINNET_TOKEN_LIST,
          },
          [
            { address: '0x6b3595068778dd592e39a122f4f5a5cf09c90fe2' }, // UNI
            { address: NATIVE_TOKEN.address },
            { address: '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984' }, // USDC
            { address: '0xdac17f958d2ee523a2206206994597c13d831ec7' }, // USDT
          ],
          {
            [CHAIN_IDS.MAINNET]: new Set(
              Object.keys(STATIC_MAINNET_TOKEN_LIST),
            ),
          },
          CHAIN_IDS.MAINNET,
        ),
      mockStore,
    );
    // The first 10 tokens returned
    const first10Tokens = [...result.current(() => true)].slice(0, 10);
    expect(first10Tokens).toMatchSnapshot();
  });

  it('should not return tokens that are not in the allowlist', () => {
    const mockStore = createBridgeMockStore({
      metamaskStateOverrides: {
        completedOnboarding: true,
        allDetectedTokens: {
          '0x1': {
            '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': [
              {
                address: '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984',
                decimals: 6,
              }, // USDC
            ],
          },
        },
      },
    });
    const { result } = renderHookWithProvider(
      () =>
        useTokensWithFiltering(
          {
            [NATIVE_TOKEN.address]: NATIVE_TOKEN,
            ...STATIC_MAINNET_TOKEN_LIST,
          },
          [
            { address: '0x6b3595068778dd592e39a122f4f5a5cf09c90fe2' }, // UNI
            { address: NATIVE_TOKEN.address },
            { address: '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984' }, // USDC
            { address: '0xdac17f958d2ee523a2206206994597c13d831ec7' }, // USDT
          ],
          // Only 1 token in allowlist
          {
            [CHAIN_IDS.MAINNET]: new Set([
              '0x6b3595068778dd592e39a122f4f5a5cf09c90fe2',
            ]),
          },
          CHAIN_IDS.MAINNET,
        ),
      mockStore,
    );
    // The first 5 tokens returned
    const first5Tokens = [...result.current(() => true)].slice(0, 5);
    expect(first5Tokens).toMatchSnapshot();
  });
});
