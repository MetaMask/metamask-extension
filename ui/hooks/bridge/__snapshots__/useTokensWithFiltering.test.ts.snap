// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTokensWithFiltering should not return tokens that are not in the allowlist 1`] = `
[
  {
    "address": "0x0000000000000000000000000000000000000000",
    "balance": "0.000000000000000014",
    "chainId": "0xe708",
    "decimals": 18,
    "image": "./images/eth_logo.svg",
    "string": "0.000000000000000014",
    "symbol": "ETH",
    "tokenFiatAmount": 3.53395e-14,
    "type": "NATIVE",
  },
  {
    "address": "0x0000000000000000000000000000000000000000",
    "balance": "0.00000000000000001",
    "chainId": "0x1",
    "decimals": 18,
    "image": "./images/eth_logo.svg",
    "string": "0.00000000000000001",
    "symbol": "ETH",
    "tokenFiatAmount": 2.5242500000000003e-14,
    "type": "NATIVE",
  },
  {
    "address": "0x6b3595068778dd592e39a122f4f5a5cf09c90fe2",
    "aggregators": [],
    "balance": "",
    "chainId": "0x1",
    "decimals": 18,
    "erc20": true,
    "erc721": false,
    "iconUrl": "images/contract/sushi.svg",
    "image": "images/contract/sushi.svg",
    "name": "SushiSwap",
    "string": undefined,
    "symbol": "SUSHI",
    "type": "TOKEN",
  },
  {
    "address": "0x0000000000000000000000000000000000000000",
    "balance": "0x0",
    "chainId": "0x1",
    "decimals": 18,
    "iconUrl": "./images/eth_logo.svg",
    "image": "./images/eth_logo.svg",
    "name": "Ether",
    "string": "0x0",
    "symbol": "ETH",
    "type": "NATIVE",
  },
  {
    "address": "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984",
    "aggregators": [],
    "balance": "",
    "chainId": "0x1",
    "decimals": 18,
    "erc20": true,
    "iconUrl": "images/contract/uni.svg",
    "image": "images/contract/uni.svg",
    "name": "Uniswap",
    "string": undefined,
    "symbol": "UNI",
    "type": "TOKEN",
  },
]
`;

exports[`useTokensWithFiltering should return all tokens when chainId === activeChainId, sorted by balance 1`] = `
[
  {
    "address": "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984",
    "balance": "0.00184",
    "chainId": "0x1",
    "decimals": 6,
    "image": undefined,
    "isNative": false,
    "string": "0.00184",
    "tokenFiatAmount": 0.004232,
    "type": "TOKEN",
  },
  {
    "address": "0x0000000000000000000000000000000000000000",
    "balance": "0.000000000000000014",
    "chainId": "0xe708",
    "decimals": 18,
    "image": "./images/eth_logo.svg",
    "string": "0.000000000000000014",
    "symbol": "ETH",
    "tokenFiatAmount": 3.53395e-14,
    "type": "NATIVE",
  },
  {
    "address": "0x0000000000000000000000000000000000000000",
    "balance": "0.00000000000000001",
    "chainId": "0x1",
    "decimals": 18,
    "image": "./images/eth_logo.svg",
    "string": "0.00000000000000001",
    "symbol": "ETH",
    "tokenFiatAmount": 2.5242500000000003e-14,
    "type": "NATIVE",
  },
  {
    "address": "0x514910771af9ca656af840dff83e8264ecf986ca",
    "balance": "1",
    "chainId": "0x1",
    "image": undefined,
    "isNative": false,
    "string": "1",
    "tokenFiatAmount": null,
    "type": "TOKEN",
  },
  {
    "address": "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984",
    "balance": "",
    "chainId": "0x1",
    "decimals": 6,
    "string": undefined,
    "type": "TOKEN",
  },
  {
    "address": "0x6b3595068778dd592e39a122f4f5a5cf09c90fe2",
    "aggregators": [],
    "balance": "",
    "chainId": "0x1",
    "decimals": 18,
    "erc20": true,
    "erc721": false,
    "iconUrl": "images/contract/sushi.svg",
    "image": "images/contract/sushi.svg",
    "name": "SushiSwap",
    "string": undefined,
    "symbol": "SUSHI",
    "type": "TOKEN",
  },
  {
    "address": "0x0000000000000000000000000000000000000000",
    "balance": "0x0",
    "chainId": "0x1",
    "decimals": 18,
    "iconUrl": "./images/eth_logo.svg",
    "image": "./images/eth_logo.svg",
    "name": "Ether",
    "string": "0x0",
    "symbol": "ETH",
    "type": "NATIVE",
  },
  {
    "address": "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984",
    "aggregators": [],
    "balance": "",
    "chainId": "0x1",
    "decimals": 18,
    "erc20": true,
    "iconUrl": "images/contract/uni.svg",
    "image": "images/contract/uni.svg",
    "name": "Uniswap",
    "string": undefined,
    "symbol": "UNI",
    "type": "TOKEN",
  },
  {
    "address": "0xdac17f958d2ee523a2206206994597c13d831ec7",
    "aggregators": [],
    "balance": "",
    "chainId": "0x1",
    "decimals": 6,
    "erc20": true,
    "iconUrl": "images/contract/usdt.svg",
    "image": "images/contract/usdt.svg",
    "name": "Tether USD",
    "string": undefined,
    "symbol": "USDT",
    "type": "TOKEN",
  },
  {
    "address": "0x0000000000000000000000000000000000000000",
    "balance": "0x0",
    "chainId": "0x1",
    "decimals": 18,
    "iconUrl": "./images/eth_logo.svg",
    "image": "./images/eth_logo.svg",
    "name": "Ether",
    "string": "0x0",
    "symbol": "ETH",
    "type": "NATIVE",
  },
]
`;
